paddle/fluid/framework/data_layout_transform.h

```diff
@@ -36,6 +36,7 @@ inline MKLDNNMemoryFormat ToMKLDNNFormat(const DataLayout& layout) {
     case DataLayout::kNHWC:
       return MKLDNNMemoryFormat::nhwc;
     case DataLayout::kNCHW:
+    case DataLayout::kAnyLayout:
```

Comment:
Why line39 here?

Reply:
That was a temporary solution introduced to  be consistent with behavior defined here:
https://github.com/PaddlePaddle/Paddle/blob/d1a6e112e60e19fcfdc0ff4ea69a5fdb5c63a1d8/paddle/fluid/framework/data_layout_transform.cc#L147-L148

which is when layout is unspecified e.g. kAnyLayout then assume it is kNCHW.

But I have just replaced this with more proper fix